src/main/java/life/qbic/qpostman/common/functions/SearchDataSets.java

```diff
@@ -58,15 +59,28 @@ public SampleSearchCriteria searchCriteria() {
         }
 
         public SampleFetchOptions fetchOptions() {
+            // for all samples fetch the direct parents with type
             SampleFetchOptions parentFetchOptions = new SampleFetchOptions();
             parentFetchOptions.withType();
+            // fetch dataset with sample, sample type and parents with type, propagate fetch options to children.
             SampleFetchOptions sampleFetchOptions = new SampleFetchOptions();
             sampleFetchOptions.withDataSets().withSample();
             sampleFetchOptions.withType();
             sampleFetchOptions.withParents();
             sampleFetchOptions.withParentsUsing(parentFetchOptions);
             sampleFetchOptions.withChildrenUsing(sampleFetchOptions);
-            return sampleFetchOptions;
+            // for the root sample, fetch all parents recursively
+            SampleFetchOptions rootParentFetchOptions = new SampleFetchOptions();
+            rootParentFetchOptions.withType();
+            rootParentFetchOptions.withParentsUsing(rootParentFetchOptions);
+            // use same fetch options as for all other samples + fetching all parents recursively instead of direct parents.
+            SampleFetchOptions rootSampleFetchOptions = new SampleFetchOptions();
+            rootSampleFetchOptions.withDataSets().withSample();
+            rootSampleFetchOptions.withType();
+            rootSampleFetchOptions.withParents();
+            rootSampleFetchOptions.withChildrenUsing(sampleFetchOptions);
+            rootSampleFetchOptions.withParentsUsing(rootParentFetchOptions);
+            return rootSampleFetchOptions;
```

Comment:
I had do draw a treeMap to hopefully understand what is happening here even with the comments. 😅  
But this makes sense GJ